h2o-py/h2o/h2o.py

```diff
@@ -2120,6 +2122,41 @@ def upload_mojo(mojo_path):
     return mojo_estimator
 
 
+def print_mojo(mojo_path, format="json", tree_index=None):
+    """
+    Generates string representation of an existing MOJO model. 
+    :param mojo_path:  Path to the MOJO archive on the user's local filesystem
+    :param format:  Output format. Possible values: json (default), dot 
+    :param tree_index:  Index of tree to print (only work dot format)
+    :return: An string representation of given MOJO in given format
+    """    
+    assert_is_type(mojo_path, str)
+    assert_is_type(format, str, None)
+    assert_satisfies(format, format in [None, "json", "dot"])
+    assert_is_type(tree_index, int, None)
+
+    ls = H2OLocalServer()
+    jar = ls._find_jar()
+    java = ls._find_java()
+    if format is None:
+        format = "json"
+    cmd = [java, "-cp", jar, "hex.genmodel.tools.PrintMojo", "--input", mojo_path, "--format", format]
+    if tree_index is not None:
+        cmd += ["--tree", str(tree_index)]
+    output_file = tempfile.mkstemp(prefix="mojo_output")[1]
+    try:
+        with open(output_file, 'w+') as stdout:
+            return_code = subprocess.call(cmd, stdout=stdout)
+            stdout.seek(0)
+            output = stdout.read()
```

Comment:
this is all Python 2.x way of handling subprocesses, most of it is deprecated.
If we don't have any generic impl in h2o that handles subprocesses for both Python 3 and 2, then we should  change that.

Reply:
especially we should ensure that the subprocess is always terminated or killed.

Reply:
same as above, we probably need to support py2 for some more time

Reply:
sure, not saying we should drop support.
But I'll create ticket to ensure that we move towards more Python3 friendliness (= hide the deprecated, sometimes buggy, Py2-specific methods or techniques).